From fcc828cdc9616457a643389387f0ca9fba12b867 Mon Sep 17 00:00:00 2001
From: Anton Blanchard <anton@ozlabs.org>
Date: Mon, 2 Aug 2021 11:02:10 +1000
Subject: [PATCH] Don't warn about stack size on host binaries
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

I'm hitting a stack size warning when building pflash:

common/arch_flash_powerpc.c: In function ‘get_dev_mtd.constprop’:
common/arch_flash_powerpc.c:177:1: error: the frame size of 8240
    bytes is larger than 2048 bytes [-Werror=frame-larger-than=]

That function has 2 PATH_MAX strings, each of which will use up 4kB of
stack.

We've tried to work around the issue of stack size warnings on host
binaries in a few places, with limited success. This patch removes the
check completely instead. We need to modify the HOSTCFLAGS variable
assignment to be immediate for this to work.

Upstream-Status: Backport [https://github.com/open-power/skiboot/commit/fcc828cdc9616457a643389387f0ca9fba12b867]
Signed-off-by: Anton Blanchard <anton@ozlabs.org>
Signed-off-by: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
---
 Makefile.main            | 5 +----
 external/pflash/rules.mk | 1 -
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/Makefile.main b/Makefile.main
index d21f27be3..189b4ae41 100644
--- a/Makefile.main
+++ b/Makefile.main
@@ -51,7 +51,7 @@ endif
 # Host tools and options
 HOSTCC=gcc
 HOSTEND=$(shell uname -m | sed -e 's/^i.*86$$/LITTLE/' -e 's/^x86.*/LITTLE/' -e 's/^ppc64le/LITTLE/' -e 's/^ppc.*/BIG/')
-HOSTCFLAGS=-O1 $(CWARNS) -DHAVE_$(HOSTEND)_ENDIAN -MMD
+HOSTCFLAGS:=-O1 $(CWARNS) -DHAVE_$(HOSTEND)_ENDIAN -MMD
 HOSTCFLAGS += $(call try-cflag,$(HOSTCC),-std=gnu11)
 HOSTCFLAGS += $(call try-cflag,$(HOSTCC),-m64)
 HOSTCFLAGS += $(call try-cflag,$(HOSTCC),-Wjump-misses-init) \
@@ -62,9 +62,6 @@ HOSTCFLAGS += $(call try-cflag,$(HOSTCC),-Wjump-misses-init) \
 HOSTCFLAGS += -DDEBUG -DCCAN_LIST_DEBUG
 
 # We want small stack usage for skiboot
-# but host compilation of unit tests tend to inline heavily,
-# which creates larger stack frames and triggering useless warnings
-HOSTCFLAGS += -Wframe-larger-than=4096
 CWARNS += -Wframe-larger-than=1024
 
 HOSTGCOVCFLAGS = -fprofile-arcs -ftest-coverage -lgcov -O0 -g -pg
diff --git a/external/pflash/rules.mk b/external/pflash/rules.mk
index 8d5a7bfd2..1d1b60486 100644
--- a/external/pflash/rules.mk
+++ b/external/pflash/rules.mk
@@ -52,7 +52,6 @@ $(LIBFLASH_OBJS): libflash-%.o : libflash/%.c | links
 $(CCAN_OBJS): ccan-list-%.o: ccan/list/%.c | links
 	$(Q_CC)$(CC) $(CFLAGS) -c $< -o $@
 
-$(EXE): CFLAGS += -Wframe-larger-than=2048
 $(EXE): $(OBJS)
 	$(Q_CC)$(CC) $(LDFLAGS) $(CFLAGS) $^ -lrt -o $@
 
-- 
2.37.1

